### DISCUSSION TITLE
Please help: how to setup telescope-select-ui

### DISCUSSION THREAD

**Discussion by convers39** [NvChad/NvChad#1576] [Q&A] [answered]
I was trying to add [telescope-select-ui](https://github.com/nvim-telescope/telescope-ui-select.nvim) for lsp code_actions.

I followed the official doc but not sure why it does not work:

```lua
-- overrides.lua
M.telescope = {
  extensions = {
    ["ui-select"] = {
    require("telescope.themes").get_dropdown {
    -- even more opts
      },
    },
    project = {
      base_dirs = {
        "~/Projects",
        -- {'~/dev/src2'},
        -- {'~/dev/src3', max_depth = 4},
        -- {path = '~/dev/src4'},
        -- {path = '~/dev/src5', max_depth = 2},
      },
      hidden_files = true, -- default: false
      theme = "dropdown",
      order_by = "asc",
    },
  },
}

``` 

```lua
-- custom/plugins/init.lua
  ["nvim-telescope/telescope-ui-select.nvim"] = {
    after = "telescope.nvim",
    -- setup = function()
    --   require("telescope").setup {
    --     extensions = {
    --       ["ui-select"] = {
    --         require("telescope.themes").get_dropdown {
    --           -- even more opts
    --         },
    --       },
    --     },
    --   }
    --   require("telescope").load_extension "ui-select"
    -- end,
    config = function()
      require("telescope").load_extension "ui-select"
    end,
  },
  ["nvim-telescope/telescope-project.nvim"] = {
    after = "telescope.nvim",
    config = function()
      require("telescope").load_extension "project"
    end,
  },

  ["nvim-telescope/telescope.nvim"] = {
    override_options = overrides.telescope,
  },

```

If I enable the select-ui in `overrides.lua` flie, the whole configuration will be disabled and all my custom settings will be gone. Then I try to put the `extensions` setup in `plugins/init.lua`, but it did not work when putting in `config`, and raised this error when putting in `setup`(which I commented out in the code).

<img width="1096" alt="image" src="https://user-images.githubusercontent.com/42597395/190958304-8b74f35d-2b87-423c-8afa-16a47ae36ecb.png">

I restarted neovim and ran`PackerSync` every time when I change those settings

Please can someone help? Really appreciate it!!!

Full config [here](https://github.com/convers39/nvchad-config)

**Comment by siduck** [comment, marked as answer]
@convers39 1st method didnt work because you are directly calling a module in the override_options table. u have to read the docs properly,example ; 

![image](https://user-images.githubusercontent.com/59060246/190969484-f6bb2dfb-7a19-4367-8061-38c77cf9db2d.png)

- Also im loading it after lspconfig cuz its useless to load it by default

# Solution 

```lua
 ["nvim-telescope/telescope-ui-select.nvim"] = {
    after = "nvim-lspconfig", 
  },

  ["nvim-telescope/telescope.nvim"] = {
    module = "telescope",
    override_options = function()
      return {
        extensions = {
          ["ui-select"] = {
            require("telescope.themes").get_dropdown {
              -- even more opts
            },
          },
        },
        extensions_list = { "themes", "terms", "ui-select" },
      }
    end,
  },
  ```
 -  And u got the module error cuz telescope is lazyloaded at its commands and ui-select tries to call telescope module, so what we do is load telscope also whenever telescope module is called!
  
  - the extensions_list is our own telescope option for its config, so u can just put name of your extension in it and it'll load
  
 - look at the default telescope config to understand it 
  
  
![image](https://user-images.githubusercontent.com/59060246/190970106-ee707610-4b96-4732-8961-ed617b51738e.png)

**Reply by convers39** [reply]
Appreciate you detailed explanation!! Solved!
